src/worker/EnigmaNode.js

```diff
@@ -420,16 +422,29 @@ class EnigmaNode extends EventEmitter {
    * Dial at some protocol and delegate the handling of that connection
    * @param {PeerInfo} peerInfo ,  the peer we wish to dial to
    * @param {String} protocolName , the protocl name /echo/1.0.1
-   * @param {Function} onConnection recieves (err,connection) =>{}
+   * @param {Function} onConnection receives (err,connection) =>{}
    */
   dialProtocol(peerInfo, protocolName, onConnection) {
     if (peerInfo.id.toB58String() === this.getSelfIdB58Str()) {
       this._logger.error('[-] Error : ' + MSG_STATUS.ERR_SELF_DIAL);
-      return;
-    } else {
+    }
+    else {
       this.node.dialProtocol(peerInfo, protocolName, onConnection);
     }
   }
+  /**
+   * Dial at some protocol and delegate the handling of that connection
+   * @param {PeerInfo} peerInfo ,  the peer we wish to dial to
+   * @param {Function} onConnection receives (err,connection) =>{}
+   */
+  dial(peerInfo, onConnection) {
+    if (peerInfo.id.toB58String() === this.getSelfIdB58Str()) {
+      this._logger.error('[-] Error : ' + MSG_STATUS.ERR_SELF_DIAL);
+    }
```

Comment:
I think it's unnecessary.  
`onConnection` will get the error and can handle it from there.  
This way error handling for dial will always happen in only one place.

![Screenshot from 2019-11-19 15-01-25](https://user-images.githubusercontent.com/22766533/69148731-a295d500-0add-11ea-868e-120d08dac200.png)